pkg/client.go/mapper.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2024 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"google.golang.org/protobuf/encoding/prototext"
+
+	"github.com/abcxyz/team-link/apis/v1alpha3"
+	tltypes "github.com/abcxyz/team-link/internal"
+	"github.com/abcxyz/team-link/pkg/github"
+	"github.com/abcxyz/team-link/pkg/groupsync"
+)
+
+type GoogleGroupToGitHubMapper struct {
+	GoogleGroupToGitHubTeam map[string][]string
+}
+
+// AllGroupIDs returns the set of groupIDs being mapped (the key set).
+func (g *GoogleGroupToGitHubMapper) AllGroupIDs(ctx context.Context) ([]string, error) {
+	res := make([]string, 0)
+	for key := range g.GoogleGroupToGitHubTeam {
```

Comment:
Maybe use maps.Keys() here? And also sort the output for determinism. Using the new iterator stuff it could look like:

```
return slices.Sort(slices.Collect(maps.Keys(g.GoogleGroupToGithubTeam)))
```

Reply:
return sorted result, I am hesitating on using `maps.Keys` since it's still in `exp` package.